michelson.md

```diff
@@ -1021,27 +1021,43 @@ The `COMPARE` instruction is defined over all comparable datatypes.
 We define `COMPARE` in terms of a `#DoCompare` function.
 
 ```k
-  syntax Int ::= #DoCompare(Data, Data) [function]
-
-  rule #DoCompare(V, V) => 0
-
-  rule #DoCompare(B1:Bool, B2:Bool)
-    =>       #if B1 ==Bool B2 #then  0
-       #else #if B1           #then  1
-       #else                        -1
-       #fi #fi
-
-  rule #DoCompare(I1:Int, I2:Int)
-    =>       #if I1  <Int I2 #then -1
-       #else #if I1 ==Int I2 #then  0
-       #else                        1
-       #fi #fi
-
-  rule #DoCompare(S1:String, S2:String)
-    =>       #if S1  <String S2 #then -1
-       #else #if S1 ==String S2 #then  0
-       #else                           1
-       #fi #fi
+  syntax Int ::= #DoCompare(Data, Data)                    [function]
+               | #DoCompareBool(Bool, Bool)                [function, functional]
+               | #DoCompareInt(Int,Int)                    [function, functional]
+               | #DoCompareStr(String, String)             [function, functional]
+               | #DoCompareAddr(Address, Address)          [function, functional]
+               | #DoCompareBytes(Bytes, Bytes)             [function, functional]
+               | #DoCompareKeyHash(KeyHash, KeyHash)       [function, functional]
+               | #DoCompareTimestamp(Timestamp, Timestamp) [function, functional]
+               | #DoCompareMutez(Mutez,Mutez)              [function, functional]
+
+  rule #DoCompare(V1:Bool, V2:Bool)           => #DoCompareBool(V1,V2)
+  rule #DoCompare(V1:Int, V2:Int)             => #DoCompareInt(V1,V2)
+  rule #DoCompare(V1:String, V2:String)       => #DoCompareStr(V1,V2)
+  rule #DoCompare(V1:Address, V2:Address)     => #DoCompareAddr(V1,V2)
+  rule #DoCompare(V1:Bytes, V2:Bytes)         => #DoCompareBytes(V1,V2)
+  rule #DoCompare(V1:KeyHash, V2:KeyHash)     => #DoCompareKeyHash(V1,V2)
+  rule #DoCompare(V1:Timestamp, V2:Timestamp) => #DoCompareTimestamp(V1,V2)
+  rule #DoCompare(V1:Mutez, V2:Mutez)         => #DoCompareMutez(V1,V2)
```

Comment:
Mark `[concrete]`?

Reply:
`#Ceil (#DoCompare(V1, V2)) => #Top requires (isBool(V1) andBool isBool(V1)) orBool (isStringV1) andBool isString(V1))`

Reply:
I found the #Ceil rule approach + the simpler rules works with the existing test harness --- but I have not checked the dexter proofs yet. Somehow, CI for this branch is down. I will wait until it comes back up and let CI check it.

Reply:
I think CI only runs when you merge to master, this branch is merging with the entrypoints one.